packages/mui-joy/src/SvgIcon/SvgIconProps.ts

```diff
@@ -0,0 +1,82 @@
+import { OverridableStringUnion, OverrideProps } from '@mui/types';
+import * as React from 'react';
+import { SxProps } from '../styles/defaultTheme';
+import { SvgIconClasses } from './svgIconClasses';
+
+export interface SvgIconPropsSizeOverrides {}
+
+export interface SvgIconPropsColorOverrides {}
+
+export interface SvgIconTypeMap<P = {}, D extends React.ElementType = 'svg'> {
+  props: P & {
+    /**
+     * Node passed into the SVG element.
+     */
+    children?: React.ReactNode;
+    /**
+     * Override or extend the styles applied to the component.
+     */
+    classes?: Partial<SvgIconClasses>;
+    /**
+     * The color of the component. It supports those theme colors that make sense for this component.
+     * You can use the `htmlColor` prop to apply a color attribute to the SVG element.
+     * @default 'inherit'
+     */
+    color?: OverridableStringUnion<
+      'inherit' | 'primary' | 'neutral' | 'warning' | 'success' | 'danger' | 'info',
+      SvgIconPropsSizeOverrides
```

Comment:
This should be `ColorPaletteProp` instead of hard-coded colors. You can take a look at Joy `ButtonProps.ts`.